### ISSUE TITLE
The Guice injector is not propagated alongside the OntopModelConfiguration inheritance hierarchy

### ISSUE THREAD

**Issue by samdus** [ontop/ontop#793]
### Description

The Guice injector is not propagated alongside the OntopModelConfiguration inheritance hierarchy. My comprehension of the code makes me think it differs from the desired behaviour. More precisely, the method `it.unibz.inf.ontop.injection.impl.OntopModelConfigurationImpl#getInjector` seems to expect that an injectorSupplier exists when the inheritance hierarchy is forked to support multiple inheritance. It might be part of the inheritance design, but I think that having some configuration objects that create their own injector might introduce some unexpected behaviours.

### Steps to Reproduce

1. Create an object `config`, instance of `OntopSQLOWLAPIConfiguration` 
2. Verify that `config.getInjector()` returns the same injector instance as `config.mappingOWLConfiguration.getInjector()`
3. It is not the same object because the `OntopMappingOntologyConfigurationImpl` class cannot be constructed using a `Supplier<Injector>`

**Expected behavior:** [What you expect to see]
The injector is always the same in every multiple-inheritance branch.

**Actual behavior:** [What actually happens]
The injector is different for some multiple inheritance branches.

**Reproduces how often:** [Does it happen every time?]
It happens every time.

### Attached material

A test reproduces this behaviour; see: https://github.com/samdus/ontop/blob/bugfix/configClassesShouldUseInjectorSuppliers/test/docker-tests/src/test/java/it/unibz/inf/ontop/docker/Issue793Test.java

### Versions

Current "version5" branch (as of 2024-02-23, commit f5ed79196073a9869d1450d2f10f8c5fe48c8418).

**Comment by bcogrel** [comment]
Thanks for reporting the issue and proposing a fix! The implementation should now be robust with it. 

Out of curiosity, have you experienced some bugs?

**Comment by samdus** [comment]
Thank you for your time spent on this review. It's a pleasure to be able to contribute!

I did not experience bugs within the reach of Ontop, per se. However, I found the behaviour when trying to extend Ontop in creating a new Configuration extending `OntopSQLOWLAPIConfiguration`. The extension was updating the properties to change a default implementation for an interface. This implementation had an injected constructor argument of a type defined in the new `Configuration` class. Therefore, Guice successfully created the instance at first. But, when a class instantiated by a `Configuration` on another inheritance branch needed the instance, it tried to recreate it and threw an Exception.

I hope this explanation is understandable. Let me know if you'd like more details. It's not the most trivial thing to explain!
